features/networking/multus.feature

```diff
@@ -606,3 +606,252 @@ Feature: Multus-CNI related scenarios
+      | resource | node                    |
+      | name     | <%= cb.nodes[0].name %> |
+      | key_val  | test=worker1            |
+    Then the step should succeed
+    
+    Given I register clean-up steps:
+    """
+    When I run the :label admin command with:
+      | resource | node                    |
+      | name     | <%= cb.nodes[0].name %> |
+      | key_val  | test-                   |
+    Then the step should succeed
+    """ 
+    #Labeing another worker node to make sure 3rd future pod to be scheduled on this node only
+    When I run the :label admin command with:
+      | resource | node                    |
+      | name     | <%= cb.nodes[1].name %> |
+      | key_val  | test=worker2            |
+    Then the step should succeed
+    
+    Given I register clean-up steps:
+    """
+    When I run the :label admin command with:
+      | resource | node                    |
+      | name     | <%= cb.nodes[1].name %> |
+      | key_val  | test-            	   |
+    Then the step should succeed
+    """ 
+    #Creating first pod in vlan 100
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/generic_multus_pod_nodeselector.yaml" replacing paths:
+      | ["metadata"]["name"] | pod1-vlan100 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"]| bridgevlan100 |
+      | ["spec"]["nodeSelector"]["test"] | worker1 |
+    Then the step should succeed
+    And the pod named "pod1-vlan100" becomes ready
+    And evaluation of `pod.name` is stored in the :pod1 clipboard
+    And I execute on the pod:
+      | ifconfig | net1 |
+    Then the step should succeed
+    And evaluation of `@result[:response].match(/\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}/)[0]` is stored in the :pod1_net1_ip clipboard
+    
+    #Creating 2nd pod on same node as first in vlan 100
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/generic_multus_pod_nodeselector.yaml" replacing paths:
+      | ["metadata"]["name"] | pod2-vlan100 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"]| bridgevlan100 |
+      | ["spec"]["nodeSelector"]["test"] | worker1 |
+    Then the step should succeed
+    And the pod named "pod2-vlan100" becomes ready
+    And evaluation of `pod.name` is stored in the :pod2 clipboard
+    And I execute on the pod:
+      | ifconfig | net1 |
+    Then the step should succeed
+    And evaluation of `@result[:response].match(/\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}/)[0]` is stored in the :pod2_net1_ip clipboard
+    
+    #Clean-up required to erase bridge interfaces created due to above pods on same node
+    Given I register clean-up steps:
+    """
+    the bridge interface named "mybridge" is deleted from the "<%= cb.nodes[0].name %>" node
+    the bridge interface named "mybridge.100" is deleted from the "<%= cb.nodes[0].name %>" node
```

Comment:
clean-up steps can still be moved up, immediately before or after pod is created. There should be no harm inusing these steps even if the resource have never been created. The parameters here are bridge name and node name, and these are known even before pod is created.

Something that I'm more concerned with, I think I commented in another PR about this, is that we create and delete cluster global resources like VLANs and bridges. We delete here the bridges. And they have a hardcoded name.
So my assumption is, in case multiple scenarios run simultaneously (what we do) then they will interfere with each other, likely resulting in the whole test run to be terminated breaking the whole pipeline.

If scenarios need to run one after the other, they need to be tagged `@destructive`. If dynamic names are used that make sure no interference happens, then `@destructive` is not needed and we run scenarios in parallel.

So please review scenarios here and in your other PRs that they will not interfere with each other if run simultaneously.

@pruan-rht and @liangxia can help you understand `destructive` and clean-up process if I am not available to review or answer in the following weeks.

Reply:
@akostadinov Yes i did mark them as destructive already yesterday as i also evaluated them as destructive in case. And i have adjusted the cleanup properly. The problem with cleanup is bridges does get create when a pod gets created not a net-attach-def. That's why i have defined cleanup at the end of all pods creation. But for readability i have moved them under specific net-attach-def and pod for now